### ISSUE TITLE
Unable to set up wallet password listeners: missing address manager namespace

### ISSUE THREAD

**Issue by ysangkok** [lightningnetwork/lnd#3612]
### Background

Setting up a fresh regtest wallet, I get this error.

```
$ ./lnd-debug --bitcoin.active --bitcoin.regtest --bitcoin.node=bitcoind --bitcoind.zmqpubrawblock=localhost:40000 --bitcoind.zmqpubrawtx=40000 --bitcoind.rpcuser=user --bitcoind.rpcpass=pass --bitcoind.rpchost=localhost:8332
2019-10-18 07:27:36.915 [WRN] LTND: open /home/janus/.lnd/lnd.conf: no such file or directory
2019-10-18 07:27:36.916 [INF] LTND: Version: 0.8.0-beta commit=v0.8.0-beta-18-g93247d0304260c8cfa6af5b19988a55ed5e39f14, build=development, logging=default
2019-10-18 07:27:36.919 [INF] LTND: Active chain: Bitcoin (network=regtest)
2019-10-18 07:27:36.922 [INF] CHDB: Checking for schema update: latest_version=11, db_version=11
2019-10-18 07:27:36.953 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8080
2019-10-18 07:27:36.955 [INF] RPCS: password RPC server listening on 127.0.0.1:10009
2019-10-18 07:27:36.964 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.
2019-10-18 07:28:46.366 [ERR] LTND: Could not unload new wallet: wallet is not loaded
2019-10-18 07:28:46.367 [ERR] LTND: Unable to set up wallet password listeners: missing address manager namespace
2019-10-18 07:28:46.368 [INF] LTND: Shutdown complete
Unable to set up wallet password listeners: missing address manager namespace
```

### Your environment

* version of `lnd`

commit 93247d0304260c8cfa6af5b19988a55ed5e39f14

* which operating system (`uname -a` on *Nix)

Ubuntu on WSL on Windows 10

* version of `btcd`, `bitcoind`, or other backend

bitcoind 0.18.1

* any other relevant environment details

### Steps to reproduce

Seems like you just need to create a fresh wallet to hit this.

**Comment by guggero** [comment]
What command did you run from `lncli` to get this output? Did you run `lncli create` and then `lncli unlock` later or is this error directly from the `create` command?

It looks like the `wallet.db` file is corrupt. Or it cant' be read. Can you check that it exists and that you have write permission to it?

**Comment by ysangkok** [comment]
`wallet.db` doesn't exist before, and it happens right after I do `create`. Interestingly, `lnd` is still usable because I can `unlock` (and it works) if I shut down lnd and restart. So the wallet was somehow created correctly on disk.

**Comment by guggero** [comment]
WSL is supported since it's basically just an Ubuntu (or other distro) in a VM.
But maybe the user directories are set up a bit different. Can you make sure that the home directory (of the user inside WSL) is writable and has no `.lnd` folder in it before you issue the `create` command?
Is the WSL user home directory shared with the Windows host system? Maybe some networking/sharing driver is causing issues...

**Comment by wpaulino** [comment]
That namespace is created right before it's being accessed, so it could point to an issue with the disk.

**Comment by ysangkok** [comment]
@wpaulino How do you want me to test if my disk is all right? I doubt it is defective but it is hard to prove the absence of intermittent hardware issues.

Anyway here is a more detailed test run on commit a469d5dc2cc69df005e25b505a27638cbce93fd4:

showing that `~/.lnd` does not exist:
```
janus@electrum-ux305ca:~/go/src/github.com/lightningnetwork/lnd$ ls -l ~/.lnd
ls: cannot access '/home/janus/.lnd': No such file or directory
```

showing that `~/.lnd` has usable permissions after creating with mkdir:

```
janus@electrum-ux305ca:~/go/src/github.com/lightningnetwork/lnd$ ls -ld ~/.lnd
drwxrwxrwx 1 janus janus 512 Oct 24 22:42 /home/janus/.lnd
```

now i delete the directory and start lnd using the command in the initial bug report.

i create the wallet in a parallel WSL window:
```

janus@electrum-ux305ca:~$ ~/go/bin/lncli create
Input wallet password:
Confirm password:


Do you have an existing cipher seed mnemonic you want to use? (Enter y/n): n

Your cipher seed can optionally be encrypted.
Input your passphrase if you wish to encrypt it (or press enter to proceed without a cipher seed passphrase):

Generating fresh cipher seed...

!!!YOU MUST WRITE DOWN THIS SEED TO BE ABLE TO RESTORE THE WALLET!!!

---------------BEGIN LND CIPHER SEED---------------
 1. abstract   2. fog        3. laptop   4. hub
 5. time       6. canyon     7. weasel   8. during
 9. tiny      10. service   11. invest  12. uncle
13. canvas    14. december  15. crush   16. grass
17. february  18. town      19. olive   20. key
21. family    22. faculty   23. pill    24. whale
---------------END LND CIPHER SEED-----------------

!!!YOU MUST WRITE DOWN THIS SEED TO BE ABLE TO RESTORE THE WALLET!!!

lnd successfully initialized!
```

the original lnd window now contains the error in the original report, i will not repost that.

some additional version info:

```
C:\Users\janus>wsl -l
Windows Subsystem for Linux Distributions:
Ubuntu (Default)
```

from inside WSL:

```
janus@electrum-ux305ca:~$ dpkg -l golang-1.13-go
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                     Version           Architecture      Description
+++-========================-=================-=================-=====================================================
ii  golang-1.13-go           1.13.1-1longsleep amd64             Go programming language compiler, linker, compiled st
```

**Comment by ysangkok** [comment]
@guggero 

> Is the WSL user home directory shared with the Windows host system?

I guess it is not? I don't remember being asked this question, but it can't the same directory, since inside WSL:
```
janus@electrum-ux305ca:~$ ls
db  electrumx  go
```

and inside cmd.exe:
```

C:\Users\janus>dir
 Volume in drive C is OS
 Volume Serial Number is 1C19-857A

 Directory of C:\Users\janus

24/10/2019  22:34    <DIR>          .
24/10/2019  22:34    <DIR>          ..
24/09/2019  17:19    <DIR>          .android
25/09/2019  21:31                22 .bashrc
16/10/2019  20:36            11,923 .bash_history
09/10/2019  05:30                99 .bash_profile
19/10/2019  03:47    <DIR>          .config
13/10/2019  19:39    <DIR>          .dotnet
24/10/2019  22:34    <DIR>          .eclair
19/09/2019  05:09    <DIR>          .emacs.d
23/09/2019  15:48                16 .emulator_console_auth_token
15/10/2019  17:34                74 .gitconfig
16/10/2019  22:56    <DIR>          .idlerc
19/10/2019  03:40               154 .lesshst
16/09/2019  18:39    <DIR>          .nuget
21/09/2019  20:31    <DIR>          .ssh
25/09/2019  22:15    <DIR>          .templateengine
01/04/2019  22:53    <DIR>          .vim
16/10/2019  19:25            36,152 .viminfo
09/10/2019  06:29    <DIR>          3D Objects
09/10/2019  06:29    <DIR>          Contacts
18/10/2019  08:06    <DIR>          Desktop
12/10/2019  06:33    <DIR>          Documents
24/10/2019  22:33    <DIR>          Downloads
24/10/2019  21:16    <DIR>          Dropbox
09/10/2019  06:29    <DIR>          Favorites
09/10/2019  06:29    <DIR>          Links
09/10/2019  06:29    <DIR>          Music
20/12/2017  07:33    <DIR>          OneDrive
09/10/2019  06:29    <DIR>          Pictures
20/12/2017  09:23    <DIR>          Roaming
09/10/2019  06:29    <DIR>          Saved Games
09/10/2019  06:29    <DIR>          Searches
16/10/2019  23:01    <DIR>          source
09/10/2019  06:29    <DIR>          Videos
               7 File(s)         48,440 bytes
              28 Dir(s)  20,730,232,832 bytes free

C:\Users\janus>
```

**Comment by wpaulino** [comment]
Do you run into this issue when creating the node outside of WSL, i.e. native Windows?

**Comment by ysangkok** [comment]
No, the issue is not present in native Windows using the latest official binaries.

**Comment by molxyz** [comment]
@guggero 

> WSL is supported since it's basically just an Ubuntu (or other distro) in a VM.

WSL doesn't have a real linux kernel.  I haven't tried to run LND with bitcoind on WSL but I did try to run c-lightning with bitcoind on testnet and c-lightning could never get up to the current block height. This was over a year ago. But since then I concluded that WSL can't handle a lightning node like c-lightning which is lighter than lnd then WSL can't handle lnd which can consume more resources.  I've seen developers talk about WSL as an inferior system, not a real linux system, more like a "bastardized" item that Microsoft throws on there for developers who want to compile software for linux. I wouldn't recommend anyone to use it to run LND. 

But MS now has WSL2 with a real linux kernel, I don't have it, haven't tried it but I assume it might do a better job for lnd.  https://docs.microsoft.com/en-us/windows/wsl/wsl2-about .

**Comment by guggero** [comment]
Seems there is a pretty significant difference between WSL 1 and WSL 2:

From https://wiki.ubuntu.com/WSL:
> WSL(1) runs Linux binaries by implementing a Linux API compatibility layer partly in the Windows kernel while WSL 2 uses the Linux kernel itself in a lightweight VM to provide better compatibility with native Linux installations. 

So my assumption was on something I must've read about WSL 2.

Something else I found while googling:
https://askubuntu.com/questions/992399/what-are-the-limitations-of-the-windows-subsystem-for-linux:
> The most significant limitation of WSL is that the Linux and Windows filesystems are not really compatible. Linux files use a different permissions scheme than Windows files, and when you edit or save a file in the WSL home folder ~/ you save it with metadata that will make Windows just show an empty file in Notepad. There are also some problems with the difference in line endings (LF on Linux vs CRLF on Windows) but this is much less of an issue, especially since Microsoft has enabled VS Code and Notepad to use either line ending.

To summarize: Not being an expert on WSL by any means, I would tentatively declare that we don't support running lnd on WSL 1, given that it's not recommended to do low-level file system and networking operations on an emulated kernel interface.
This might look different for WSL 2 but further tests will need to verify this.

@ysangkok given your specific problem: Can you please take a look at the second link here and see if that gives you an idea what you could try w.r.t. to file system permissions and file names? Seems to be a known problem that files can appear empty.

**Comment by guggero** [comment]
Closing this for now as it appears to be a problem of WSL file system interactions and not lnd. Feel free to reopen if there is evidence pointing towards a problem in lnd.

**Comment by sako0938** [comment]
In case it helps someone else, I got this issue running LND 0.10.1 on a Raspberry Pi Zero W, without swap space enabled. Something in the wallet creation step maxes out memory and causes something to fail and not write to disk. Happened 2x without swap enabled, didn't happen with swap enabled.

**Comment by lukechilds** [comment]
I'm running into this same issue when running LND in a Parralels VM on a shared folder via Vagrant.

**Comment by guggero** [comment]
Did you give the VM enough RAM and/or swap space? The scrypt operation involved in the wallet encryption allocates a lot of memory, as @sako0938 pointed out.

**Comment by lukechilds** [comment]
Oh that's a great point. It has 2GB which I assumed would be enough but let me try again with more.

**Comment by lukechilds** [comment]
Hmmn, I don't think it's a memory issue. I tried with 8GB RAM and 1GB swap and hit the same issue. Kept an eye on memory usage in the VM and it stayed below 1GB.

Still hitting

```
unable to set up wallet password listeners: missing address manager namespace
```

Next start of LND gives:

```
2021-11-02 07:51:56.647 [INF] LNWL: Opened wallet
2021-11-02 07:51:56.676 [INF] CHRE: Primary chain is set to: bitcoin
2021-11-02 07:51:56.679 [INF] LNWL: Started listening for bitcoind block notifications via ZMQ on 10.21.21.8:28332
2021-11-02 07:51:56.679 [INF] LNWL: Started listening for bitcoind transaction notifications via ZMQ on 10.21.21.8:28333
2021-11-02 07:51:57.129 [INF] LNWL: The wallet has been unlocked without a time limit
2021-11-02 07:51:57.130 [INF] CHRE: LightningWallet opened
2021-11-02 07:51:57.132 [INF] SRVR: Proxying all network traffic via Tor (stream_isolation=false)! NOTE: Ensure the backend node is proxying over Tor as well
panic: page 11 already freed

goroutine 207 [running]:
go.etcd.io/bbolt.(*freelist).free(0x400034cf00, 0x7, 0xffff68737000)
 go.etcd.io/bbolt@v1.3.5-0.20200615073812-232d8fc87f50/freelist.go:175 +0x398
go.etcd.io/bbolt.(*node).spill(0x4000234d20, 0x4000bf9608, 0x4)
 go.etcd.io/bbolt@v1.3.5-0.20200615073812-232d8fc87f50/node.go:359 +0x348
go.etcd.io/bbolt.(*Bucket).spill(0x4000487ec0, 0x4000afa300, 0x4000bf98d8)
 go.etcd.io/bbolt@v1.3.5-0.20200615073812-232d8fc87f50/bucket.go:570 +0x340
go.etcd.io/bbolt.(*Bucket).spill(0x40002aa558, 0xdf93f4d61, 0x1b2ba40)
 go.etcd.io/bbolt@v1.3.5-0.20200615073812-232d8fc87f50/bucket.go:537 +0x2b8
go.etcd.io/bbolt.(*Tx).Commit(0x40002aa540, 0x6012afca590b1a11, 0x3a4fc328bf5beb43)
 go.etcd.io/bbolt@v1.3.5-0.20200615073812-232d8fc87f50/tx.go:160 +0xa8
github.com/btcsuite/btcwallet/walletdb/bdb.(*transaction).Commit(0x4000178130, 0x4000178130, 0x0)
 github.com/btcsuite/btcwallet/walletdb@v1.3.6-0.20210803004036-eebed51155ec/bdb/db.go:101 +0x2c
github.com/btcsuite/btcwallet/walletdb/bdb.(*db).Update(0x4000670000, 0x4000afa330, 0x10fb3a0, 0x0, 0x0)
 github.com/btcsuite/btcwallet/walletdb@v1.3.6-0.20210803004036-eebed51155ec/bdb/db.go:437 +0x108
github.com/btcsuite/btcwallet/walletdb.Update(0x12761f0, 0x4000670000, 0x4000afa330, 0x0, 0x0)
 github.com/btcsuite/btcwallet/walletdb@v1.3.6-0.20210803004036-eebed51155ec/interface.go:273 +0x44
github.com/btcsuite/btcwallet/wallet.(*Wallet).syncWithChain(0x40004285a0, 0x4000397640, 0x12634d8, 0x4000afc150)
 github.com/btcsuite/btcwallet@v0.12.1-0.20210826004415-4ef582f76b02/wallet/wallet.go:414 +0x7f8
github.com/btcsuite/btcwallet/wallet.(*Wallet).handleChainNotifications(0x40004285a0)
 github.com/btcsuite/btcwallet@v0.12.1-0.20210826004415-4ef582f76b02/wallet/chainntfns.go:115 +0xac4
created by github.com/btcsuite/btcwallet/wallet.(*Wallet).SynchronizeRPC
 github.com/btcsuite/btcwallet@v0.12.1-0.20210826004415-4ef582f76b02/wallet/wallet.go:207 +0x288
```

Seems like some sort of issue with the Parallels VM filesystem or potentially the shared folder functionality messing something up.

For context I'm trying to get the Umbrel development tooling working on Apple silicon. Same setup works on VirtualBox. So this is likely an issue with Parallels not LND. But if there's any way LND can handle this better or you have any ideas on how I can work around the issue it would be hugely appreciated 🙏

**Comment by guggero** [comment]
Hmm... weird. I assume you did start with a fresh wallet this time? Or could this be some weird left-over state from a previous try?

The `bbolt` database does some memory mapping magic to access the file system directly. Perhaps that's just not supported by the file system in question? Or maybe there are options you can specify to make it behave differently?
Or skip `bbolt` altogether by running a Postgres DB container?

**Comment by lukechilds** [comment]
>Hmm... weird. I assume you did start with a fresh wallet this time? Or could this be some weird left-over state from a previous try?

Yep, nuked LND data dir before retrying. Just did it again to make sure.

Just tested and it works inside the Parallels VM without a shared folder, it's specifically when bi-directional sharing with the host is enabled that LND stops working. I guess just buggy behaviour with Parallels shared folders, hopefully we can transition to VMWare for when it's released for Apple sillicon.

Thanks for your help @guggero!

**Comment by sandipndev** [comment]
From some debugging, it will not happen if Swap file system is enabled.

**Comment by krtk6160** [comment]
> From some debugging, it will not happen if Swap file system is enabled.

Thanks a lot! This solved the problem for me.

**Comment by openoms** [comment]
Got hit by this also when deploying LND with kubernetes - helm on a ZFS filesystem on Ubuntu 22.04 without swap:
```
[INF] LTND: Attempting automatic wallet unlock with password provided in file
[ERR] LTND: Shutting down because error in main method: error creating wallet config: error unlocking wallet with password from file: missing address manager namespace
[INF] LTND: Shutdown complete

error creating wallet config: error unlocking wallet with password from file: missing address manager namespace
```
checked there is no swap configured:
```
cat /proc/swaps
Filename				Type		Size		Used		Priority
```
Installed zram 
```
sudo apt install zram-config
```
and working after reboot:
```
cat /proc/swaps
Filename				Type		Size		Used		Priority
/dev/zram0                              partition	8129088		0		5

zramctl
NAME       ALGORITHM DISKSIZE DATA COMPR TOTAL STREAMS MOUNTPOINT
/dev/zram0 lzo-rle       7.8G   4K   74B   12K       4 [SWAP]
```

LND now deploys without errors.
